File: server/src/routes/drift.routes.ts
import { Router } from 'express';
import { authMiddleware } from '../middlewares/auth.middleware.js';
import {
  analyzeDriftController,
  deleteDriftAnalysisController,
  getDriftAnalysisController,
  listProjectDriftAnalysesController,
  saveDriftAnalysisController,
} from '../controllers/drift.controller.js';

export const driftRouter = Router();

driftRouter.use(authMiddleware);
driftRouter.post('/analyze', analyzeDriftController);
driftRouter.post('/save', saveDriftAnalysisController);
driftRouter.get('/project/:projectId', listProjectDriftAnalysesController);
driftRouter.get('/:driftAnalysisId', getDriftAnalysisController);
driftRouter.delete('/:driftAnalysisId', deleteDriftAnalysisController);

File: server-go/internal/modules/drift/drift_routes.go
package drift

import (
	"driftledger/server-go/internal/config"
	"driftledger/server-go/internal/middleware"
	"github.com/gin-gonic/gin"
	"go.mongodb.org/mongo-driver/mongo"
)

func RegisterRoutes(group *gin.RouterGroup, db *mongo.Database, cfg config.Config) {
	handler := NewHandler(NewService(db, NewInferenceClient(cfg)))
	group.POST("/analyze-direct", handler.AnalyzeDirect)
	group.Use(middleware.Auth(db, cfg))
	group.POST("/analyze", handler.Analyze)
	group.POST("/save", handler.Save)
	group.GET("/project/:projectId", handler.List)
	group.GET("/:driftAnalysisId", handler.Get)
	group.DELETE("/:driftAnalysisId", handler.Delete)
}

func RegisterModelRoutes(group *gin.RouterGroup, db *mongo.Database, cfg config.Config) {
	handler := NewHandler(NewService(db, NewInferenceClient(cfg)))
	group.POST("/analyze", handler.AnalyzeModel)
}
